addons/mail/models/mail_message.py

```diff
@@ -833,6 +833,7 @@ def toggle_message_starred(self):
         self.env.user._bus_send(
             "mail.message/toggle_star", {"message_ids": [self.id], "starred": starred}
         )
+        return Store(self).get_result()
```

Comment:
We don't really need all message fields, do we? Limiting to just `starred` would be better.